File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long ans = 0 , cnt = 0;
        for(int x : nums) {
            if(x == 0) {
                cnt++;
                ans += cnt;
            }
            else {
                cnt = 0;
            }
        }
        return ans;
    }
};

File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long cnt = 0, curr = 0;
        for (int x : nums) {
            curr = (x == 0) ? curr + 1 : 0;
            cnt += curr;
        }
        return cnt;
    }
}